sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -193,6 +196,9 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
     case OneRowRelation =>
       ""
 
+    case Hint(_, _, child) =>
```

Comment:
so why do we have this case? `SQLBuilder` will be applied on analyzed plan right?

And I think it's better to push this check in `CheckAnalysis`